### ISSUE TITLE
Report mixed multiline signatures style

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#1937]
# Rule request

## Thesis

```python
# good, all arguments on the same line
def f1(arg1: int, arg2: int, arg3: int, arg4: int):
    return arg1 + arg2 + arg3 + arg4

# good, all arguments on the same line
def f2(
    arg1: int, arg2: int, arg3: int, arg4: int,
):
    return arg1 + arg2 + arg3 + arg4

# good, every argument on a separate line
def f3(
    arg1: int,
    arg2: int,
    arg3: int,
    arg4: int,
):
    return arg1 + arg2 + arg3 + arg4

# bad, some arguments on a separate line, some on the same
def f4(
    arg1: int, arg2: int,
    arg3: int,
    arg4: int,
):
    return arg1 + arg2 + arg3 + arg4

```

## Reasoning

For short one-line signatures, the reader will read it from left to right. For long multiline signatures, it will be read vertically. However, a mixed approach means that second arguments on every line have a chance to be missed during visual scanning. The issue becomes apparent when type annotations are longer:

```python
def func(
    arg: Union[List[int], Tuple[int], None], 
    argument: Optional[List[int]], arg2: int,
    arg3: List[int],
    arg4: List[int],
):
    ...
```

Here, it's easy to `arg2` thinking that it is a part of a long signature.

**Comment by sobolevn** [comment]
We already do that! 👍 

```python
def f4(
    arg1: int, arg2: int,
    arg3: int,
    arg4: int,
):
    return arg1 + arg2 + arg3 + arg4
```

Produces:

```
a.py

  1:1      WPS317 Found incorrect multi-line parameters
  def f4(
  ^

Full list of violations and explanations:
https://wemake-python-stylegui.de/en/0.15.2/pages/usage/violations/

```
